src/main/java/com/google/sps/firestore/FirestoreUtil.java

```diff
@@ -0,0 +1,60 @@
+package com.google.sps;
+
+import static com.google.sps.Constants.PROJECT_ID;
+
+import com.google.api.core.ApiFuture;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.FieldValue;
+import com.google.cloud.firestore.Firestore;
+import com.google.firebase.cloud.FirestoreClient;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.FirebaseOptions;
+import java.io.IOException;
+import javax.servlet.ServletException;
+
+/** Util methods for Firestore. */
+public class FirestoreUtil {
+
+  private static Firestore db;
+  private FirestoreUtil() {}
+
+  /**
+  * Gets the database instance for this project. Initializes database if it
+  * has not been initialized.
+  * @return Firestore database instance
+  * @throws IOException if database initialization fails.
+  */
+  public static Firestore getDatabase() {
+    if (db == null) {
+      try {
+        db = initializeDatabase();
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
+    }
+    return db;
+  }
+
+  /**
+  * Sets the database instance for this project.
+  * Use for setting database to be a mock database for unit testing.
+  */
+  public static void setDatabase(Firestore database) {
```

Comment:
If this is really only for testing, the name should reflect that - something like setDatabaseForTest, so it's clear that if you are using it non-test code you are doing something wrong. Also would reduce the scope and mark @VisibleForTesting

Reply:
Added annotation and changed name.